app/controllers/authentication_controller.rb

```diff
@@ -0,0 +1,14 @@
+class AuthenticationController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  def validate_otp
+    mobile_number = params[:mobile_number]
+    otp = params[:otp]
+    service = AuthenticationService.new(mobile_number, otp)
+    response = service.process
+    if response.error.present?
+      render json: { errors: [response.error] }, status: :unprocessable_entity
+    else
+      render json: { data: { token: response.token } }, status: :ok
+    end
```

Comment:
I think separate variable assignments for `mobile_number` and `otp` are unnecessary here. We can directly use `params[:mobile_number]` and `params[:otp]` when initializing the AuthenticationService.

```suggestion
    service = AuthenticationService.new(params[:mobile_number], params[:otp])
    response = service.process

    if response.error.present?
      render json: { errors: [response.error] }, status: :unprocessable_entity
    else
      render json: { data: { token: response.token } }, status: :ok
    end
```

Reply:
done